glean-core/src/traits/datetime.rs

```diff
@@ -4,6 +4,8 @@
 
 #![allow(clippy::too_many_arguments)]
 
+use crate::ErrorType;
```

Comment:
Can't comment on unmodified lines, so put it here:

We should move the `fn set` function above the `set_with_details` one, so it's the first thing visible in the file/in the docs.

I wonder ... do we really need to make that `set_with_details` part of the _exposed_ API (the trait) or can we put it on the type only (thus making it more "internal", but FOG can still use it)?
What do you think?

Reply:
> 
> 
> Can't comment on unmodified lines, so put it here:
> 
> We should move the `fn set` function above the `set_with_details` one, so it's the first thing visible in the file/in the docs.

Good idea!

> I wonder ... do we really need to make that `set_with_details` part of the _exposed_ API (the trait) or can we put it on the type only (thus making it more "internal", but FOG can still use it)?
> What do you think?

Your call :-) But why do we need this in FOG? And can this be a follow-up?

Reply:
Right now for the C++ API.
We _could_ leave it out, but then FOG would need to replicate [what [`glean-core`] is already doing](https://github.com/mozilla/glean/blob/141ae1da58b9da794ebbee5ac34f189fb9b4ff6d/glean-core/src/metrics/datetime.rs#L83-L94).

Reply:
I'm torn on this. On one hand, this is already done by glean-core, as you said. On the other, we're adding API surface just to not have FOG copy paste these lines :)

I'd be fore removing the API completely and have FOG copy/paste that :D But it's really your call.

Reply:
Then I call it: let's remove that API! (... wait for chutten to complain)